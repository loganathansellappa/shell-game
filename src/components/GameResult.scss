.result-container {
  display: flex;
  flex-direction: column;
  //justify-content: space-around;
  align-items: center;
  flex-wrap: nowrap;
  font-size: 100px;

  .winner {
    animation: shake 1s;
    animation-iteration-count: 2;
  }

  .loser {
    animation: shake 0.1s;
    animation-iteration-count: 5;
  }

  @keyframes shake {
    0% {
      transform: rotate(0deg);
    }
    25% {
      transform: rotate(5deg);
    }
    50% {
      transform: rotate(0eg);
    }
    75% {
      transform: rotate(-5deg);
    }
    100% {
      transform: rotate(0deg);
    }
  }
  color: #cccccc;
  .winner {
    color: #cccccc;
  }
  .loser {
    color: #cccccc;
  }
}
