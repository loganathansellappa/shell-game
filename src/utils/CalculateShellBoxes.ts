import { Children } from 'react';

const calculateShellBoxes = (children) => {
  const boundingBoxes = {};

  Children.forEach(children, (child) => {
    const domNode = child.ref.current;
    if (domNode) {
      const nodeBoundingBox = domNode.getBoundingClientRect();

      boundingBoxes[child.key] = nodeBoundingBox;
    }
  });

  return boundingBoxes;
};

export default calculateShellBoxes;
