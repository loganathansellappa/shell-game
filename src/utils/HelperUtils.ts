import { ColorCodes } from '~/@types/GenericTypes';

export const shellCount = (): number => Number(process.env.SHELL_COUNT);

export const shuffle = <T>(elements: T[]): Array<T> => {
  for (let i = elements.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [elements[i], elements[j]] = [elements[j], elements[i]];
  }
  return [...elements];
};

export const getColors = (count: number): ColorCodes => {
  let colorCodes = {};
  for (let i = 0; i < count; i++) {
    colorCodes[`key${i}`] = `#${Math.floor(Math.random() * 16777215).toString(
      16
    )}`;
  }
  return colorCodes;
};

export enum Level {
  Beginner = 2,
  Intermediate = 4,
  Expert = 8,
}
