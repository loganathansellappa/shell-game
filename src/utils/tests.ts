import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};
export const testStore = () => {
  const mockStore = configureStore();
  const initialState = {
    gameLog: {
      gameCount: 1,
      winners: 1,
      losers: 0,
    },
  };
  const store = mockStore(initialState);
  return store;
};
