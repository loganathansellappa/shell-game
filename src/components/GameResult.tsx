import './GameResult.scss';

interface GameResultProps {
  winner: boolean;
  reset: () => void;
}

const GameResult = ({ winner, reset }: GameResultProps) => {
  let text = "You've Won";
  if (!winner) {
    text = "You've lose! Try again";
  }
  return (
    <div className={'result-container'}>
      <p className={winner ? 'winner' : 'loser'}>{text}</p>
      <button onClick={() => reset()}>Reset</button>
    </div>
  );
};

export default GameResult;
