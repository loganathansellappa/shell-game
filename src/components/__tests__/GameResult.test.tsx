import GameResult from '../GameResult';
import { renderWithRouter } from '../..//utils/tests';

test('renders GameResult Component', () => {
  const { container } = renderWithRouter(
    <GameResult winner={false} reset={jest.fn()} />
  );
  expect(container.firstChild).toMatchSnapshot();
  expect(container.textContent).toContain("You've lose! Try again");
});
