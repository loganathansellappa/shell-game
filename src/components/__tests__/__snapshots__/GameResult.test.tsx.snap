// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GameResult Component 1`] = `
<div
  class="result-container"
>
  <p
    class="loser"
  >
    You've lose! Try again
  </p>
  <button>
    Reset
  </button>
</div>
`;
